google-cloud-spanner/lib/google/cloud/spanner/data.rb

```diff
@@ -151,16 +151,7 @@ def to_a
         #
         def to_h
           raise DuplicateNameError if fields.duplicate_names?
-          hashified_pairs = pairs.map do |key, value|
-            if value.is_a? Data
-              [key, value.to_h]
-            elsif value.is_a? Array
-              [key, value.map { |v| v.is_a?(Data) ? v.to_h : v }]
-            else
-              [key, value]
-            end
-          end
-          Hash[hashified_pairs]
```

Comment:
Also nice that it eliminates the duplicate conversion code!

Reply:
I don't get why it is faster this way, but it is.